### PR TITLE
feat: Check vc status

### LINKED ISSUE #76: Check vc status

### PR DESCRIPTION
**Pull request by fqutishat**
closes #76

Signed-off-by: Firas Qutishat <firas.qutishat@securekey.com>

### CODE REVIEW

File: pkg/restapi/vc/operation/operations.go [hunk 1]
```diff
@@ -69,17 +77,26 @@ type Client interface {
...
+func New(provider storage.Provider, client Client, kms legacykms.KMS, vdri vdriapi.Registry,
+	hostURL string) (*Operation, error) {
 	store, err := provider.OpenStore(credentialStore)
 	if err != nil {
 		return nil, err
 	}
 
+	c := crypto.New(kms, verifiable.NewDIDKeyResolver(vdri))
+
+	vcStatusManager, err := cslstatus.New(provider, hostURL, 50, c)
```
Comment (sandrask) [inline pkg/restapi/vc/operation/operations.go]: List size should be configurable, or at least constant

File: pkg/restapi/vc/operation/operations.go [hunk 2]
```diff
@@ -143,10 +161,39 @@ func (o *Operation) verifyCredentialHandler(rw http.ResponseWriter, req *http.Re
...
+		}
+
+		for _, vcStatus := range csl.VC {
+			if !strings.Contains(vcStatus, vc.ID) {
+				continue
+			}
+
+			statusVc, err := o.parseAndVerifyVC([]byte(vcStatus))
+			if err != nil {
+				o.writeErrorResponse(rw, http.StatusBadRequest,
```
Comment (sandrask) [inline pkg/restapi/vc/operation/operations.go]: InternalServerError

File: pkg/restapi/vc/operation/operations.go [hunk 3]
```diff
@@ -143,10 +161,39 @@ func (o *Operation) verifyCredentialHandler(rw http.ResponseWriter, req *http.Re
...
 
-	_, err = o.parseAndVerifyVC(body)
+	vc, err := o.parseAndVerifyVC(body)
 	if err != nil {
 		verified = false
 		message = err.Error()
+	} else {
+		csl, err := o.vcStatusManager.GetCSL(vc.Status.ID)
+		if err != nil {
+			o.writeErrorResponse(rw, http.StatusBadRequest,
```
Comment (sandrask) [inline pkg/restapi/vc/operation/operations.go]: regular error message

File: pkg/restapi/vc/operation/operations.go [hunk 4]
```diff
@@ -143,10 +161,39 @@ func (o *Operation) verifyCredentialHandler(rw http.ResponseWriter, req *http.Re
 	verified := true
 	message := successMsg
 
-	_, err = o.parseAndVerifyVC(body)
+	vc, err := o.parseAndVerifyVC(body)
 	if err != nil {
 		verified = false
 		message = err.Error()
```
Comment (sandrask) [inline pkg/restapi/vc/operation/operations.go]: return here; separate between check VC and check VC status (with comment)

File: pkg/restapi/vc/operation/operations.go [hunk 5]
```diff
@@ -143,10 +161,39 @@ func (o *Operation) verifyCredentialHandler(rw http.ResponseWriter, req *http.Re
...
+				fmt.Sprintf("failed to get credential status list: %s", err.Error()))
+			return
+		}
+
+		for _, vcStatus := range csl.VC {
+			if !strings.Contains(vcStatus, vc.ID) {
+				continue
+			}
+
+			statusVc, err := o.parseAndVerifyVC([]byte(vcStatus))
```
Comment (sandrask) [inline pkg/restapi/vc/operation/operations.go]: Create new function verifyVCStatus

Comment (fqutishat) [reply pkg/restapi/vc/operation/operations.go]: done

File: pkg/restapi/vc/operation/operations.go (hunk 3 above)
Comment (fqutishat) [reply pkg/restapi/vc/operation/operations.go]: done

File: pkg/restapi/vc/operation/operations.go (hunk 2 above)
Comment (fqutishat) [reply pkg/restapi/vc/operation/operations.go]: done

File: pkg/restapi/vc/operation/operations.go (hunk 1 above)
Comment (fqutishat) [reply pkg/restapi/vc/operation/operations.go]: done

**Pull request merged**